src/main/scala/au/com/agiledigital/toolform/model/Resource.scala

```diff
@@ -9,17 +9,23 @@ import com.typesafe.config.Config
   * Developers can provided mappings between a resource and the external resource in their 'dev-resources' file. This
   * will convert the Resource into a MappedResource, which is a Composable element.
   *
-  * @param id           the id of the resource.
-  * @param resourceType the type of the resource (e.g. postgres, elastic search)
-  * @param settings     additional settings for the resource.
-  * @param image        the docker URL for the image associated with this resource.
-  * @param environment  a map of environment variables to inject into the container.
-  * @param exposedPorts a list of ports that are exposed to an outside network.
-  *                     (This is usually done through the edge/nginx reverse proxy but sometimes you need to expose a container)
+  * @param id            the id of the resource.
+  * @param resourceType  the type of the resource (e.g. postgres, elastic search)
+  * @param settings      additional settings for the resource.
+  * @param image         the docker URL for the image associated with this resource.
+  * @param environment   a map of environment variables to inject into the container.
+  * @param exposedPorts a list of ports that are exposed to other services in the internal network.
+  * @param externalPorts  a list of ports that are exposed to an outside network.
+  *                      (This is usually done through the edge/nginx reverse proxy but sometimes you need to expose a container)
   */
-final case class Resource(id: String, resourceType: String, settings: Option[Config], image: String, environment: Map[String, String] = Map(), exposedPorts: List[String] = List())
-    extends ProjectElement
-    with Service {
+final case class Resource(id: String,
+                          resourceType: String,
+                          settings: Option[Config],
+                          image: String,
+                          environment: Map[String, String] = Map(),
+                          exposedPorts: List[String] = List(),
```

Comment:
`List` -> `immutable.Seq`, `List()` -> `Nil`.

Reply:
Is `immutable.Seq` preferable to `Nil` in this case?